### ISSUE TITLE
[research] converting public keys to addresses as authentication keys for ECDSA, ECDSA Recoverable and Schnorr validators.

### ISSUE THREAD

**Issue by feliciss** [rooch-network/rooch#616]
Presently, Ed25519 scheme encodes Rooch default public keys to addresses as authentication keys to the account authentication.

While mentioning other schemes like ECDSA over Secp256k1: https://github.com/rooch-network/rooch/pull/610#discussion_r1290842394, it's whether we can encode Ethereum public keys to Ethereum addresses and use them as authentication keys to the Rooch account authentication.

Some bit researches I did, is that we can use `ecdsa_k1_validator.move` for Bitcoin public keys to addresses using the SHA-256 and RIPEMD-160 hash to apply from, and why are we using ECDSA over Secp256k1 is that the signature length in `ecdsa_k1_validator.move` matches Bitcoin's traditional DER-encoded ECDSA 64-byte signature length, while the signature length in `ecdsa_k1_recoverable_validator.move` matches Ethereum's 65-byte recoverable R, S, V signature length.

So the same way, we can and should use `ecdsa_k1_recoverable_validator.move` for converting public keys to Ethereum addresses as authentication keys.

For Schnorr signature, it can be applied for converting Schnorr public keys to Nostr addresses as authentication keys for Nostr-like applications, although it's not in a high priority.

In conclusion, in four ways, presumably we can apply for public keys to address, separately for schemes of validator:

1. ECDSA over Secp256k1 validator: converting public keys to Bitcoin addresses as authentication keys for Bitcoin applications.
2. ECDSA Recoverable Signature over Secp256k1 validator: converting public keys to Ethereum addresses as authentication keys for Ethereum L1 and L2 applications.
3. Schnorr validator: converting public keys to Nostr addresses as authentication keys for Nostr applications.
4. Ed25519 validator: converting public keys to Rooch addresses as authentication keys for Rooch L2.
